repos/spack_repo/builtin/packages/gaudi/package.py

```diff
@@ -57,6 +57,19 @@ class Gaudi(CMakePackage, CudaPackage):
 
     maintainers("drbenmorgan", "vvolkl", "jmcarcell")
 
+    _cxxstd_values = (
+        conditional("14", when="@:39"),
+        conditional("17", when="@:39"),
+        conditional("20", when="@39:"),
+    )
+    _cxxstd_common = {
+        "values": _cxxstd_values,
+        "multi": False,
+        "description": "Use the specified C++ standard when building.",
+    }
+    variant("cxxstd", default="17", when="@:39", **_cxxstd_common)
+    variant("cxxstd", default="20", when="@39:", **_cxxstd_common)
```

Comment:
Version ranges in spack are inclusive at the higher end. Did you mean:
```suggestion
    variant("cxxstd", default="17", when="@:38", **_cxxstd_common)
    variant("cxxstd", default="20", when="@39:", **_cxxstd_common)
```
These are otherwise conflicting defaults. Also see the definition of `_cxxstd_values` above.